SDWebImage/Core/SDWebImageManager.m

```diff
@@ -277,7 +291,7 @@ - (void)callDownloadProcessForOperation:(nonnull SDWebImageCombinedOperation *)o
                 [self callCompletionBlockForOperation:operation completion:completedBlock error:error url:url];
             } else if (error) {
                 [self callCompletionBlockForOperation:operation completion:completedBlock error:error url:url];
-                BOOL shouldBlockFailedURL = [self shouldBlockFailedURLWithURL:url error:error];
+                BOOL shouldBlockFailedURL = [self shouldBlockFailedURLWithURL:url error:error options:options context:context];
                 
```

Comment:
Previously, `SDWebImageRetryFailed` had been forgotten ?

Reply:
Actually, it's the old logic, which I don't think correct.
See: https://github.com/SDWebImage/SDWebImage/blob/4.4.8/SDWebImage/SDWebImageManager.m#L222 (4.x version)

The old logic is: When we download failed with a 404 for example, even we provude `SDWebImageRetryFailed`, the URL will be added into the black list. This means, if next time we load the same URL without that `SDWebImageRetryFailed`, we will get a immediatelly callback with error.

The new logic is: When we download failed with a 404 for example, when we provide `SDWebImageRetryFailed`, it will never been added to black list. This means, if next time we load the same URL without that  that `SDWebImageRetryFailed`, we will actually try to send HTTP request, but this time, because it does not have `SDWebImageRetryFailed`, it will be added to black list.

I think the new logic is better than the old one. But if you think this effect some of logic, I can revert back that one.

Reply:
Revert back this change to the previous version. This time we do not change this behavior. In the future we will design a better black list system.